### ISSUE TITLE
Bookmark game

### ISSUE THREAD

**Issue by veloce** [lichess-org/mobile#594]
Add the possibility to bookmark a game.

Should be available in context menu (long press), in the future game list screen, and in a game screen.

**Comment by veloce** [comment]
Yes indeed the bookmark action is not opened to oAuth. Will create an issue on Lila. Thanks for the report.

**Comment by veloce** [comment]
Server PR: https://github.com/lichess-org/lila/pull/15726

**Comment by veloce** [comment]
The `bookmark` filter should be added to the game list endpoint: https://lichess.org/api#tag/Games/operation/apiGamesUser

So we can display bookmarked games.

**Comment by veloce** [comment]
@ZTL-UwU see: https://github.com/lichess-org/lila/pull/15726#issuecomment-2228088127

**Comment by ZTL-UwU** [comment]
@veloce One more problem, there is currently no way to know if a game is bookmarked by myself or not.

I think there are potentially two solutions to this.
1. We can tweek the `/game/export/$id` api and add a response field to indicate whether the game is bookmarked by the requesting user.
2. We can probably cache the bookmarked game list (using https://lichess.org/api#tag/Games/operation/apiGamesUser) and check if the game is included in the list or not. This is a bit trickier to implement tho.

**Comment by veloce** [comment]
So it misses the `bookmarked` boolean field in the game json. I'll see to add it server side.

**Comment by ZTL-UwU** [comment]
> So it misses the `bookmarked` boolean field in the game json. I'll see to add it server side.

Actually, maybe all the game related apis will need the bookmarked field because the bookmark button is in almost all game screens. e.g. the analysis screen, game screen, archived game screen and the game history screen.
